### PR TITLE
[ML] Delete job document

### PR DESCRIPTION
**Pull request by davidkyle**
Marks the job as deleting then at the end of the delete job process the job document is removed. 

The checks that were in `MlMetadata.deleteJob` (checking the job is not open and no datafeed references it) are made in the action now. However, it is not simple to keep the [check](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlMetadata.java#L290) that only a job marked as deleting can be deleted. Given that the job can only be deleted from this action deleting a job not marked for deletion cannot happen

### CODE REVIEW

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
I pushed another commit to make the intention clearer when deleting the job document. That is I'm saying don't error if the doc has already been removed by a concurrent delete process.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java [hunk 1]
```diff
@@ -208,16 +208,37 @@ public void onFailure(Exception e) {
     }
 
     /**
-     * Delete the anomaly detector job config document
+     * Delete the anomaly detector job config document.
+     * {@code errorIfMissing} controls whether or not an error if return
+     * if the document does not exist.
      *
      * @param jobId The job id
+     * @param errorIfMissing If the job document does not exist and this this true
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java]: typo: this this

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java [hunk 2]
```diff
@@ -208,16 +208,37 @@ public void onFailure(Exception e) {
     }
 
     /**
-     * Delete the anomaly detector job config document
+     * Delete the anomaly detector job config document.
+     * {@code errorIfMissing} controls whether or not an error if return
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java]: typo: if return -> is returned

**Review by droberts195 — APPROVED** [review]
LGTM apart from a couple of typos in the comments

**Pull request merged**